.github/workflows/cloud-deploy-reusable-docker.yml

```diff
@@ -0,0 +1,204 @@
+      # This value is added, so you can test the image after it's built in test mode and by doing a health-check.
+      test_port:
+        required: false
+        default: '1341'
+        type: string
+      # The boolean that helps to determine whether to perform a health check. This is not a required input and defaults to false.
+      health_check:
+        required: false
+        default: false
+        type: boolean
+      # The tag under which the image is built, it should align with the name in the project command docker:build
+      local_tag:
+        required: true
+        type: string
+      # The environment tag. Possible values are dev, stg, and prod. This is not a required input of type string.
+      env_tag:
+        required: false
+        type: string
+      enterprise:
+        required: false
+        type: boolean
+    outputs:
+      docker_image:
+        description: 'The image that was built'
+        value: ${{ jobs.reusable_docker.outputs.docker_image }}
+
+jobs:
+  job-setup:
+    runs-on: ubuntu-latest
+    outputs:
+      platforms: ${{ steps.set-platforms.outputs.platforms }}
+      dockerName: ${{ steps.set-docker-name.outputs.dockerName }}
+    steps:
+      - id: set-docker-name
+        run: |
+          if [[ "${{ inputs.enterprise }}" == "true" ]]; then
+            echo "dockerName=${{ inputs.package_name }}-ee" >> "$GITHUB_OUTPUT"
+          else
+            echo "dockerName=${{ inputs.package_name }}" >> "$GITHUB_OUTPUT"
+          fi
+
+      - id: set-platforms
+        run: |
+          if [[ "${{ inputs.environment }}" == "Production" ]]; then
+            echo "platforms=linux/amd64" >> "$GITHUB_OUTPUT"
+          else
+            echo "platforms=linux/amd64" >> "$GITHUB_OUTPUT"
+          fi
+
+  reusable_docker:
+    needs: [job-setup]
+    runs-on: ubuntu-latest
+    timeout-minutes: 80
+    environment: ${{ inputs.environment }}
+    outputs:
+      docker_image: ${{ steps.save-image-to-output.outputs.IMAGE }}
+    permissions:
+      contents: read
+      packages: write
+      deployments: write
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: ${{ inputs.enterprise }}
+          token: ${{ secrets.SUBMODULES_TOKEN }}
+
+      - name: Prepare
+        id: prepare
+        shell: bash
+        run: |
+          service=${{ inputs.package_name }}
+          servicename=${service#*/} 
+          echo "SERVICE_NAME=$servicename" >> "$GITHUB_OUTPUT"
+
+      - uses: ./.github/actions/setup-project
+        with:
+          slim: 'true'
+          submodules: ${{ inputs.enterprise }}
+
+      - uses: crazy-max/ghaction-setup-docker@v2
+        with:
+          version: v24.0.6
+          daemon-config: |
+            {
+              "features": {
+                "containerd-snapshotter": true
+              }
+            }
+
+      - name: Setup QEMU
```

Comment:
@AliaksandrRyzhou Since this is about cloud deployment, why do we need QEMU?

Reply:
I've left it as an opportunity to manage platforms in the future. It didn't overload CICD significantly. I'll delete it, I agree

Reply:
Thank you. Let's only keep what's needed for now for cloud deployments. Every second counts.